spring-cloud-dataflow-shell/src/main/java/org/springframework/cloud/dataflow/shell/command/ModuleCommands.java

```diff
@@ -191,14 +195,45 @@ public Object getValue(int row, int column) {
 				int currentRow = row - 1;
 				if (mappings.get(key).size() > currentRow) {
 					return mappings.get(key).get(currentRow);
-				} else {
+				}
+				else {
 					return null;
 				}
 			}
 		};
 		return DataFlowTables.applyStyle(new TableBuilder(model)).build();
 	}
 
+	@CliCommand(value = IMPORT_APPS, help = "Register all apps listed in a properties file")
+	public String importFromResource(
+			@CliOption(mandatory = true,
+					key = {"", "uri"},
+					help = "URI for the properties file")
+			String uri,
+			@CliOption(key = "force",
+				help = "force update if any app already exists (only if not in use)",
+				specifiedDefaultValue = "true",
+				unspecifiedDefaultValue = "false")
+			boolean force) {
+		moduleOperations().importFromResource(uri, force);
+		return String.format("Successfully registered apps from '%s'", uri);
+	}
+
+	@CliCommand(value = REGISTER_APPS, help = "Register all apps provided in a map")
+	public String registerAll(
+			@CliOption(mandatory = true,
+					key = {"", "map"},
+					help = "apps as key/value pairs with key as 'type.name' and value as a URI")
+			Properties apps,
```

Comment:
The thing is, I don't really see the value of inline properties. It takes 2 registrations to make sense (otherwise it's good ol' `module register`) and to me it gets very very hard to read. So people would most certainly have those props in a file somewhere (not to mention the need to do/redo registrations), so `--propertiesFile` is the thing we should promote IMHO.

That being said, I agree that if we're gonna use the `properties/propertiesFile` terminology, then we should somehow be consistent with stream deployment, and use a comma as separator.